packages/async-flow/src/replay-membrane.js

```diff
@@ -571,7 +571,7 @@ export const makeReplayMembrane = ({
           }
         }
       },
-    );
+    ).catch(reject);
```

Comment:
This really shouldn't be necessary, we should never have a rejection at this point. We're likely papering over a much larger bug.

Reply:
@erights could you help me figure this one out?

Reply:
What I found out so far:

An error is being *throw* from then body of the async `when` function defined at the bottom of `when.js`

The error is
```
Error: chain info lacks staking denom
    at makeError (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/assert.js:347:61)
    at fail (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/assert.js:479:20)
    at Fail (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/ses/src/error/assert.js:489:39)
    at file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/orchestration/src/exos/remote-chain-facade.js:98:25
    at asVow (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/vow/src/vow-utils.js:90:16)
    at Object.makeAccount (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/orchestration/src/exos/remote-chain-facade.js:94:18)
    at In "makeAccount" method of (RemoteChainFacade public) [as makeAccount] (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/node_modules/@endo/exo/src/exo-tools.js:171:24)
    at performCall (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/async-flow/src/replay-membrane.js:136:30)
    at guestCallsHost (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/async-flow/src/replay-membrane.js:197:19)
    at In "makeAccount" method of (RemoteChainFacade public) [as makeAccount] (file:///Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/async-flow/src/replay-membrane.js:466:16)
    at <anonymous> (/Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/orchestration/test/facade.test.ts:121:33) {stack: 'Error: chain info lacks staking denom
    at …ges/orchestration/test/facade.test.ts:121:33)', message: 'chain info lacks staking denom'}
```

The `hVow` argument is a vow for a promise rejected with this error. 

The `when` function is *not* calling the `onRejected` handler with this error.

What I don't yet understand is why the `when`  body throws this error *instead of* passing it to the `onRejected` callback.

attn @michaelfig ?

Reply:
Exactly what I found. This is a bug in vow's `when` I'm fixing now.

Reply:
Got it, I think a bug in the `when` implementation. PR coming shortly if I'm right.

Reply:
I have the PR and test incoming